src/ConsoleBuffer/Commands/ControlSequence.cs

```diff
@@ -54,17 +73,22 @@ protected override void Parse(string bufferData)
                 startIndex = 1;
             }
 
-            this.Parameters = bufferData.Substring(startIndex).Split(new[] { ';' });
-        }
+            var idx = startIndex;
+            var len = bufferData.Length;
 
-        protected int ParameterToNumber(int offset, int defaultValue = 0, int maxValue = short.MaxValue)
-        {
-            if (this.Parameters.Count > offset && ushort.TryParse(this.Parameters[offset], out var paramValue))
+            // Parses the string in bufferData to extract the int's represented by text,
+            // it will do it without an allocations.  bufferData takes the form of
+            // NUM;NUM;NUM... for an unknown number of of NUM's
+            while (idx < len)
             {
-                return Math.Max(defaultValue, Math.Min(paramValue, maxValue));
+                var val = 0;
+                while (idx < len && bufferData[idx] != ';')
+                {
+                    val = val * 10 + (bufferData[idx++] - '0');
```

Comment:
This is smart and clever code if you only expect integers or ';', but it doesn't actually check the contents of the bufferData and bail out if it's not within the expected bounds..

Additionally, 0-9 is the most common so that's a reasonable assumption given the existing codebase, however there are subtle issues. Looking at http://invisible-island.net/xterm/ctlseqs/ctlseqs.html in the 'CSI Pm m' command explainer:
```
          o   ISO-8613-6 has been interpreted in more than one way;
              xterm allows the semicolons separating the subparameters
              in this control to be replaced by colons (but after the
              first colon, colons must be used).
```
meaning there is at least one more allowable non ';' value. That said this is a bit of quibbling and could probably be ignored in the scope of CSI sequences. Just worth mentioning.

In any event I think this section needs to be much more robust in handling characters that are not digits (0-9) or the ';' separator, and if you change it then unit test changes seem appropriate as well.

Reply:
The spec says it can either be a ';' or ':', but the first one found wins.  Since this is probably constrained to cmd.exe, wsl.exe and powershell.exe, it's probably not going to be ':' all of a sudden.

Reply:
Unit tests added.

Reply:
Thanks! I think ignoring the ':' stuff is fine for now although it may need to be addressed in future.